### DISCUSSION TITLE
Sparse Bitmap32

### DISCUSSION THREAD

**Discussion by lamdalili** [graphics32/graphics32#323] [General] [answered]
I'm using `TBitmapLayer` to draw and display paths my question is  how to save in the Bitmap32 only  the areas generated by the path, in other words making a kind of Bitmap32 that allocates memory only for the used pixels and ignore the transparent parts

![pl](https://github.com/user-attachments/assets/f389960f-9b72-4c78-81a4-b212693d50ee)

**Comment by andersmelander** [comment, marked as answer]
I'm afraid that's not possible; A bitmap is always a single contiguous block of memory backing a single rectangular area.

It would be possible to create a tile system (like that of PhotoShop & Gimp) on top of TBitmap32 but that feature belongs in the application domain and is beyond the scope of a low level library like Graphics32.

If you are drawing paths and are concerned about the memory usage of bitmap layers, I suggest you instead create a custom layer type that simply draws the associated paths on the back buffer on-demand. Derive your layer from `TPositionedLayer` and override the `Paint` method.

**Comment by lamdalili** [comment]
Thanks  for answering

> If you are drawing paths and are concerned about the memory usage of bitmap layers, I suggest you instead create a custom layer type that simply draws the associated paths on the back buffer on-demand. Derive your layer from TPositionedLayer and override the Paint method.

The program looked something like that before the introduction of TBitmapLayer to speed up the paint operation which was very slow especially when the number of paths is large (freezing when scrolling) and  also I need it to detect hittest on the path.

The goal was to find a way to optimize memory usage for Bitmap32 which in some cases doesn't exceed 5% of the total size ... but it seems this is not possible as expected.
